src/UserCollectableAttribute.js

```diff
@@ -234,14 +190,17 @@ function mixinIdentifiers(UCA) {
     const { identifier } = def;
 
     function UCAConstructor(value, version) {
-      const self = new UserColectableAttribute(identifier, value, version);
+      const self = new UserCollectableAttribute(identifier, value, version);
       return self;
     }
 
     source[name] = UCAConstructor;
-    _.mixin(UserColectableAttribute, source);
+    _.mixin(UserCollectableAttribute, source);
   });
   return UCA;
 }
 
-module.exports = mixinIdentifiers(UserColectableAttribute);
+const UserCollectableAttributeToExport = mixinIdentifiers(UserCollectableAttribute);
+UserCollectableAttributeToExport.resolveType = resolveType;
+
+module.exports = UserCollectableAttributeToExport;
```

Comment:
Why do this?

Reply:
This is done in order to 'mixin' the UserCollectableAttribute to have constructor like this:
`new UCA.NameGivenNames('John')`